rs/ethereum/evm-rpc-client/src/types.rs

```diff
@@ -103,6 +103,45 @@ pub mod candid {
         pub topics: Option<Vec<Vec<String>>>,
     }
 
+    #[derive(Clone, Debug, PartialEq, Eq, CandidType, Deserialize)]
+    pub struct FeeHistoryArgs {
+        /// Number of blocks in the requested range.
+        /// Typically, providers request this to be between 1 and 1024.
+        #[serde(rename = "blockCount")]
+        pub block_count: u128,
+
+        /// Highest block of the requested range.
+        /// Integer block number, or "latest" for the last mined block or "pending", "earliest" for not yet mined transactions.
+        #[serde(rename = "newestBlock")]
+        pub newest_block: BlockTag,
+
+        /// A monotonically increasing list of percentile values between 0 and 100.
+        /// For each block in the requested range, the transactions will be sorted in ascending order
+        /// by effective tip per gas and the corresponding effective tip for the percentile
+        /// will be determined, accounting for gas consumed.
+        #[serde(rename = "rewardPercentiles")]
+        pub reward_percentiles: Option<Vec<u8>>,
+    }
+
+    #[derive(Debug, Clone, Serialize, Deserialize, PartialEq, CandidType)]
+    pub struct FeeHistory {
+        /// Lowest number block of the returned range.
+        #[serde(rename = "oldestBlock")]
+        pub oldest_block: Nat,
+        /// An array of block base fees per gas.
+        /// This includes the next block after the newest of the returned range,
+        /// because this value can be derived from the newest block.
+        /// Zeroes are returned for pre-EIP-1559 blocks.
+        #[serde(rename = "baseFeePerGas")]
+        pub base_fee_per_gas: Vec<Nat>,
+        /// An array of block gas used ratios (gasUsed / gasLimit).
+        #[serde(default)]
+        #[serde(rename = "gasUsedRatio")]
+        pub gas_used_ratio: Vec<f64>,
```

Comment:
Did you consider scaling `gasUsed` so that you can use `Nat` here instead of `f64`?

Reply:
No I didn't because this type should be the same as the one used by the EVM-RPC canister, which uses floating points in the Candid [interface](https://github.com/internet-computer-protocol/evm-rpc-canister/blob/main/candid/evm_rpc.did#L56) and in the underlying Rust [implementation](https://sourcegraph.com/github.com/dfinity/ic@5991b07d0e396b7885d5b4d4c917f7290dd70814/-/blob/rs/ethereum/cketh/minter/src/eth_rpc.rs?L429)

Reply:
Okay, in that case, using floating-point numbers makes sense, I guess. Thanks!